src/template.cpp

```diff
@@ -225,37 +259,53 @@ std::string template_t::render_sections(const std::string& tmplate,
         {
             show = "true";
         }
+        
         // inverted section?
         if (modifier == "^" && show == "false") show = "true";
         else if (modifier == "^" && show == "true") show = "false";
-        // assign replacement content
+        
+        // Generate content
         if (show == "true")
         {
-            if (boost::regex_search(matches[3].first, matches[3].second, section,
-                                    boost::match_default | boost::format_all))
-            {
-                repl.assign(template_t::render_sections(matches[3], ctx));
-            }
-            else
+            for(CollectionType::iterator it = values.begin();
+                it != values.end(); ++it)
             {
-                for(CollectionType::iterator it = values.begin();
-                    it != values.end(); ++it)
+                std::string content = matches[3];
+                Context small_ctx;
+                small_ctx = ctx;
+                small_ctx.add(*it);
+                
+                if (std::regex_search(content, section,
+                                        std::regex_constants::match_default | std::regex_constants::format_default))
                 {
-                  Context small_ctx;
-                  small_ctx = ctx;
-                  small_ctx.add(*it);
-                  repl += template_t::render_tags(matches[3], small_ctx);
+                    content.assign(template_t::render_sections(content, small_ctx));
                 }
+                
+                repl += template_t::render_tags(content, small_ctx);
             }
         }
+        // Hide section with empty content
         else repl.assign("");
-        ret += boost::regex_replace(text, section, repl,
-                                    boost::match_default | boost::format_all);
-        rest.assign(matches[0].second, end);
-        start = matches[0].second;
+        
```

Comment:
can you please wrap this in curly braces as well. I know it's not required but I think it makes the code clearer.